### PR TITLE
Add WWW-Authenticate HTTP response header to token-related problem types

### LINKED ISSUE #126: Support WWW-Authenticate HTTP response header
See https://github.com/belgif/rest-guide/issues/204.

Support for WWW-Authenticate HTTP response header should be added to:

* [NoAccessTokenProblem](https://github.com/belgif/rest-problem-java/blob/main/belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/NoAccessTokenProblem.java)
* [InvalidAccessTokenProblem](https://github.com/belgif/rest-problem-java/blob/main/belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/InvalidAccessTokenProblem.java)
* [ExpiredAccessTokenProblem](https://github.com/belgif/rest-problem-java/blob/main/belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/ExpiredAccessTokenProblem.java)
* [MissingScopeProblem](https://github.com/belgif/rest-problem-java/blob/main/belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/MissingScopeProblem.java)

### PR DESCRIPTION
**Pull request by jpraet**
Fixes #126

### CODE REVIEW

File: belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/InvalidAccessTokenProblem.java [hunk 1]
```diff
@@ -47,9 +51,43 @@ public class InvalidAccessTokenProblem extends ClientProblem {
 
     private static final long serialVersionUID = 1L;
 
+    private String reason = "The access token is invalid";
```
Comment (clone1612) [inline belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/InvalidAccessTokenProblem.java:54]: Field can be made final

File: belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/MissingScopeProblem.java [hunk 2]
```diff
@@ -99,4 +101,9 @@ public int hashCode() {
         return Objects.hash(super.hashCode(), requiredScopes);
     }
 
+    @Override
+    public Map<String, Object> getHttpResponseHeaders() {
+        return Collections.singletonMap(WWW_AUTHENTICATE, "Bearer error=\"insufficient_scope\"");
```
Comment (clone1612) [inline belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/MissingScopeProblem.java:106]: scope(s) to indicate you might be missing multiple?

File: belgif-rest-problem/src/test/java/io/github/belgif/rest/problem/ExpiredAccessTokenProblemTest.java [hunk 3]
```diff
@@ -18,6 +19,8 @@ void construct() {
         assertThat(problem.getStatus()).isEqualTo(401);
         assertThat(problem.getDetail())
                 .isEqualTo("The Bearer access token found in the Authorization HTTP header has expired");
+        assertThat(problem.getHttpResponseHeaders()).containsEntry(HttpResponseHeaders.WWW_AUTHENTICATE,
```
Comment (clone1612) [inline belgif-rest-problem/src/test/java/io/github/belgif/rest/problem/ExpiredAccessTokenProblemTest.java:22]: Shouldn't we also validate the header is properly set in our integration tests?

**Review by clone1612 — CHANGES_REQUESTED** [review]

File: belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/InvalidAccessTokenProblem.java (hunk 1 above)
Comment (jpraet) [reply belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/InvalidAccessTokenProblem.java:54]: Only if we make some changes to the initialization / constructor chaining logic.

How do you see this? The default constructor calling `this("The access token is invalid");`?

File: belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/MissingScopeProblem.java (hunk 2 above)
Comment (jpraet) [reply belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/MissingScopeProblem.java:106]: The error code "insufficient_scope" is part of the spec.
https://datatracker.ietf.org/doc/html/rfc6750#section-3.1

File: belgif-rest-problem/src/test/java/io/github/belgif/rest/problem/ExpiredAccessTokenProblemTest.java (hunk 3 above)
Comment (jpraet) [reply belgif-rest-problem/src/test/java/io/github/belgif/rest/problem/ExpiredAccessTokenProblemTest.java:22]: I'm not sure if this is really needed, as we already have an integration test for the Retry-After HTTP header, which uses the same .getHttpResponseHeaders() mechanism?

File: belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/InvalidAccessTokenProblem.java (hunk 1 above)
Comment (clone1612) [reply belgif-rest-problem/src/main/java/io/github/belgif/rest/problem/InvalidAccessTokenProblem.java:54]: I was thinking about:

```
    private final String reason;

    public InvalidAccessTokenProblem() {
        this(DETAIL, "The access token is invalid");
    }

    public InvalidAccessTokenProblem(String reason) {
        this(DETAIL + ": " + reason, reason);
    }

    private InvalidAccessTokenProblem(String detail, String reason) {
        super(TYPE_URI, HREF, TITLE, STATUS);
        setDetail(detail);
        this.reason = reason;
    }
```

File: belgif-rest-problem/src/test/java/io/github/belgif/rest/problem/ExpiredAccessTokenProblemTest.java (hunk 3 above)
Comment (clone1612) [reply belgif-rest-problem/src/test/java/io/github/belgif/rest/problem/ExpiredAccessTokenProblemTest.java:22]: Maybe not, I think it's personal preference. I usually treat the application as a "black box" testing all scenarios (not considering shared logic) but as you say it's tested through Retry-After.

**Review by clone1612 — APPROVED** [review]

**Pull request merged**